proposals/004-skips/README.md

```diff
@@ -0,0 +1,143 @@
+
+```yaml
+apiVersion: core.spinoperator.dev/v1alpha1
+kind: SpinApp
+metadata:
+  name: wid-spinapp
+spec:
+  image: "ghcr.io/spinkube/containerd-shim-spin/examples/spin-rust-hello:v0.13.0"
+  replicas: 1
+  executor: containerd-shim-spin
+  workloadIdentity:
+    enabled: true
+    providers:
+      azure:
+        tenantId: "9b9924f3-596a-4b18-a251-3de9eeef2bbc"
+        variableProviders:
+          - type: keyvault
+            uri: "https://my-keyvault.vault.azure.net/"
+        keyValueProviders:
+          - type: cosmosdb
+            account: "my-cosmosdb"
+            database: "my-database"
+            container: "my-container"
+    serviceAccount: "my-service-account"
+```
+
+In this case the operator would be responsible for provisioning the following Azure resources, a user-assigned managed identity in Azure, an Azure Federated Identity Credential using the cluster's OIDC issurer URL, a CosmosDB SQL role assignment for read/write to the collection, and the Key Vault access policy to read secrets. Once those Azure resources have been created, then the operator should reconcile the service account and the SpinApp deployment.
+
+The operator would then need to create the following Kubernetes resources (some details omitted for brevity):
+```yaml
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  annotations:
+    "azure.workload.identity/client-id": "YYYYYYYY-YYYY-YYYY-YYYY-YYYYYYYYYYYY" # The client ID of the user-assigned managed identity
+  name: "my-service-account"
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: wid-spinapp
+spec:
+  template:
+    metadata:
+      labels:
+        "azure.workload.identity/use": "true"
+    spec:
+      containers:
+      - name: wid-spinapp
+        image: "ghcr.io/spinkube/containerd-shim-spin/examples/spin-rust-hello:v0.13.0"
+```
+
+#### SpinApp with Azure Workload Identity using an Existing Identity
+The yaml below shows an example of a SpinApp resource that uses Azure Workload Identity using a pre-existing identity. This is useful for when the operator is not able to create identities. For example, if a company does not allow the operator rights to create identities, and identities must be created via a security team.
+
+In this case, the operator would not try to create the identity, but would instead use the existing identity and reconcele the service account and decorate the deployment with the necessary annotations and service account name.
+
+```yaml
+apiVersion: core.spinoperator.dev/v1alpha1
+kind: SpinApp
+metadata:
+  name: wid-spinapp
+spec:
+  image: "ghcr.io/spinkube/containerd-shim-spin/examples/spin-rust-hello:v0.13.0"
+  replicas: 1
+  executor: containerd-shim-spin
+  workloadIdentity:
+    enabled: true
```

Comment:
Does the user need the ability to enable/disable workload identity within the manifest? I'm wondering if the presence of `providers`, `serviceAccount`, or other fields not being `nil` is enough to consider that the user is opting into workload identity or not.

If this is the wrong place to debate the schema of the `workloadIdentity` object, let me know. :)

Reply:
You, sir, have come to the right place. Step on up to debate the schema!!

I'm indifferent about the `enabled` field. Truth be told, I deleted 1 less time than I typed it.